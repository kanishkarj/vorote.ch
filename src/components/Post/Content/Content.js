// @flow strict
import React from 'react';
import styled from 'styled-components';
import { marginTop, marginBottom } from '../../../utils/get-margin';
import { lineHeight } from '../../../utils/get-line-height';
import { mediaUp } from '../../../utils/get-media-query';

const StyledContent = styled.div`
  padding: 0 15px;
  margin: 0 auto;
  ${mediaUp.md`
    padding: 0;
  `}
`;

const ContentTitle = styled.h1`
  font-size: ${props => props.theme.fontSizes.base * 2}px;
  margin-left: auto;
  margin-right: auto;
  font-weight: ${props => props.theme.fontWeights.heading};
  text-align: center;
  ${lineHeight(1.65)}
  ${marginTop(1)}
  margin-bottom: 0;

  ${mediaUp.md`
    font-size: ${props => props.theme.fontSizes.base * 3}px;
    ${lineHeight(2.25)}
    ${marginTop(2.25)}
    ${marginBottom(1.5)}
  `}
`;

const ContentBody = styled.div`
  & figure {
    ${marginBottom(1)}
  } 

  & figure blockquote {
    font-style: italic;
    text-align: center;
    margin-top: 0;
    padding: ${props => props.theme.spacing}px 0;
  }

  & figure blockquote p {
    font-size: ${props => props.theme.fontSizes.base * 1.6817}px;
    ${lineHeight(1.5)}
    margin-top: 0;
    ${marginBottom(1)}
  }

  & a {
    text-decoration: underline;
  }

  & * {
    margin-left: auto;
    margin-right: auto
  }

  & img {
    max-width: 100%;
  }

  ${mediaUp.md`
    font-size: ${props => props.theme.fontSizes.base * 1.125}px;
    ${lineHeight(1.125)}
    ${marginBottom(1.125)}

    & p {
      font-size: ${props => props.theme.fontSizes.base * 1.125}px;
      ${lineHeight(1.125)}
      ${marginBottom(1.125)}
    }
  `}
`;

type Props = {
  body: string,
  title: string
};

const Content = ({ body, title }: Props) => (
  <StyledContent>
    <ContentTitle>{title}</ContentTitle>
    <ContentBody dangerouslySetInnerHTML={{ __html: body }} />
  </StyledContent>
);

export default Content;
