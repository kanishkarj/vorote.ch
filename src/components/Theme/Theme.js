import React from 'react';
import { ThemeProvider } from 'styled-components';

const theme = {
  fonts: {
    body: '-apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen-Sans, Ubuntu, Cantarell, "Helvetica Neue", sans-serif;',
    heading: '-apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen-Sans, Ubuntu, Cantarell, "Helvetica Neue", sans-serif;',
    monospace: 'Menlo, monospace',
  },
  fontWeights: {
    body: 400,
    heading: 600,
    bold: 700,
  },
  fontSizes: {
    base: 16,
    small: 14,
  },
  lineHeight: 1.625,
  spacing: Math.round(16 * 1.625),
  buttons: {
    height: 35,
    borderRadius: 20,
  },
  breakpoints: {
    xs: 0,
    sm: 576,
    md: 768,
    lg: 992,
    xl: 1200
  }
};

const Theme = ({ children }) => (
  <ThemeProvider theme={theme}>
    {children}
  </ThemeProvider>
);

export default Theme;
