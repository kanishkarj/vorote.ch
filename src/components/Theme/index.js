// @flow strict
export { default } from './Theme';
