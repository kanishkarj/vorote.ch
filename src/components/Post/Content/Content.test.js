// @flow strict
import React from 'react';
import renderer from 'react-test-renderer';
import Content from './Content';
import 'jest-styled-components'
import Theme from '../../Theme';

describe('Content', () => {
  it('renders correctly', () => {
    const props = {
      title: 'test',
      body: '<p>test</p>'
    };

    const tree = renderer.create(
      <Theme>
        <Content {...props} />
      </Theme>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
