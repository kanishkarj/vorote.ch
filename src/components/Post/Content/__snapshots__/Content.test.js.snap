// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content renders correctly 1`] = `
.c0 {
  padding: 0 15px;
  margin: 0 auto;
}

.c1 {
  font-size: 32px;
  margin-left: auto;
  margin-right: auto;
  font-weight: 600;
  text-align: center;
  line-height: 2.68125em;
  margin-top: 26px;
  margin-bottom: 0;
}

.c2 figure {
  margin-bottom: 26px;
}

.c2 figure blockquote {
  font-style: italic;
  text-align: center;
  margin-top: 0;
  padding: 26px 0;
}

.c2 figure blockquote p {
  font-size: 26.9072px;
  line-height: 2.4375em;
  margin-top: 0;
  margin-bottom: 26px;
}

.c2 a {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c2 * {
  margin-left: auto;
  margin-right: auto;
}

.c2 img {
  max-width: 100%;
}

@media screen and (min-width:48em) {
  .c0 {
    padding: 0;
  }
}

@media screen and (min-width:48em) {
  .c1 {
    font-size: 48px;
    line-height: 3.65625em;
    margin-top: 58.5px;
    margin-bottom: 39px;
  }
}

@media screen and (min-width:48em) {
  .c2 {
    font-size: 18px;
    line-height: 1.828125em;
    margin-bottom: 29.25px;
  }

  .c2 p {
    font-size: 18px;
    line-height: 1.828125em;
    margin-bottom: 29.25px;
  }
}

<div
  className="c0"
>
  <h1
    className="c1"
  >
    test
  </h1>
  <div
    className="c2"
    dangerouslySetInnerHTML={
      Object {
        "__html": "<p>test</p>",
      }
    }
  />
</div>
`;
